### ISSUE TITLE
compileHasher.js

### ISSUE THREAD

**Issue by TalDerei** [chnejohnson/simple-tornado#5]
What does this file do? And thanks for building this repo!

**Comment by johnson86tw** [comment]
It's for generating Hasher contract.

**Comment by TalDerei** [comment]
Right, what does the hasher contract do though, and why do we need to deploy it? much thanks!

**Comment by TalDerei** [comment]
what's the difference "circomlib/src/mimcsponge_gencontract.js" (generated by the hasher contract), and just calling the library inside of merkleTree.ts (i.e. const circomlib = require("circomlib");
const mimcsponge = circomlib.mimcsponge;

Not sure why different mimcsponge functions need to be called, rather than just using the standard library.

Thanks!

**Comment by TalDerei** [comment]
And one last question, what exactly are the solidity "Mock" contracts inside of contracts/Mock? Thanks so much for your help!
